Objects/moduleobject.c

```diff
@@ -784,11 +786,88 @@ _PyModuleSpec_IsUninitializedSubmodule(PyObject *spec, PyObject *name)
     return rc;
 }
 
+static int
+_get_file_origin_from_spec(PyObject *spec, PyObject **origin)
```

Comment:
FWIW, I often find i more clear for "return" arguments to have a "p_' prefix.  It reduces confusion when reading the code.

```suggestion
_get_file_origin_from_spec(PyObject *spec, PyObject **p_origin)
```